### ISSUE TITLE
Not an Issue : Just a question about nodes

### ISSUE THREAD

**Issue by DushanthaS** [khuedoan/homelab#148]
Hello there, awesome work. I am planning to run this on a bunch of Lenovo M720q bare metals (I have 3), However, I am wondering if I can mix and match with bare metal and VMs ( inside proxmox)? 

The idea is to take advantage of the auto-provisioning feature of bare metal and at the same time achieve HA with VMs. Is that possible?
Also is there any way to set node affinity?

**Comment by khuedoan** [comment]
Yes, you can mix and match bare metal and VMs, but it might not be the optimal solution to achieve what you want because even though 3 nodes are already enough for HA.
If you add another layer of virtualization on top of them, you still need to patch the virtualization hosts and configure HA within Proxmox. In this case, it might be better to let Kubernetes handle the node drain during reboot.

I still use Proxmox, though for a different reason: its flexibility. Even with a single node, I can create a test cluster to experiment with HA features https://github.com/khuedoan/homecloud/tree/master/staging/homelab
